### DISCUSSION TITLE
Progressive download buffering

### DISCUSSION THREAD

**Discussion by earthsojourner** [Rafostar/clapper#327] [Q&A]
Hello,

I was curious if folks new the behavior of the "Progressive download buffering" option? I often stream video files from a NAS on my local network via SMB3 and I was curious what this option does and if there is a recommendation on enabling it.

Warm regards,
Joshua

**Comment by Rafostar** [comment]
Normally GStreamer caches ahead only very small part of video into RAM. With progressive "download" buffering, as name suggests we download and cache data into temporary file from start to finish. This allows to buffer whole video on the receiving machine and seek faster in it at the cost of storing data into disk as stream goes.

Normally this option is not enabled. If you encounter network buffering issues when streaming from your local NAS or web videos, then alternatively experimental playbin3 option can be tried which has improved network buffering logic over its predecessor (playbin2).
